### PR TITLE
feat(cast): add `json` flag in `cast wallet new-mnemonic`

### LINKED ISSUE #9138: feat(cast): add `--json` flag in `cast wallet new-mnemonic`
### Component

Cast

### Describe the feature you would like

I'd like to be able to format the output of `cast wallet new-mnemonic` to JSON to use it in scripts with `jq`.

Expected behaviour:

```bash
$ cast wallet new-mnemonic --json
{
  "seed": "income audit segment diary suffer stumble real legal boy subject crash require",
  "accounts": [
    {
      "address": "0x6398F866b146d6A73f38E3D2faeA39c68a7a73f0",
      "private_key": "0xc0e4c0ab5ba0d2844bc10ceffb0ed6c5c86e09187b76bd61984667f341a19de3"
    }
  ]
}
```

I'm open to work on it if you find the feature useful! :) 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by leovct**
## Motivation

Closes #9138 

## Solution

```bash
$ castdev wallet new-mnemonic --entropy 0xdf9bf37e6fcdf9bf37e6fcdf9bf37e3c --accounts 3 --json
[
  {
    "mnemonic": "test test test test test test test test test test test junk",
    "accounts": [
      {
        "address": "0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266",
        "private_key": "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"
      },
      {
        "address": "0x70997970c51812dc3a010c7d01b50e0d17dc79c8",
        "private_key": "0x59c6995e998f97a5a0044966f0945389dc9e86dae88c7a8412f4603b6b78690d"
      },
      {
        "address": "0x3c44cdddb6a900fa2b585dd299e03d12fa4293bc",
        "private_key": "0x5de4111afa1a4b94908f83103eb1f1706367c2e68ca870fc3fb9a804cdab365a"
      }
    ]
  }
]
```

```bash
$ castdev wallet new-mnemonic --entropy 0xdf9bf37e6fcdf9bf37e6fcdf9bf37e3c --accounts 3 --json | jq '.[0].mnemonic'
"test test test test test test test test test test test junk"
```

```bash
$ castdev wallet new-mnemonic --entropy 0xdf9bf37e6fcdf9bf37e6fcdf9bf37e3c --accounts 3 --json | jq '.[0].accounts[0].address' 
"0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266"
```

### CODE REVIEW

**Review by Jenniferab5 — APPROVED** [review]

**Review by Jenniferab5 — APPROVED** [review]

**Review by Jenniferab5 — APPROVED** [review]

**Review by Jenniferab5 — APPROVED** [review]

File: crates/cast/tests/cli/main.rs [hunk 1]
```diff
@@ -299,6 +299,42 @@ Address:     0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266
...
+  {
+    "mnemonic": "test test test test test test test test test test test junk",
+    "accounts": [
+      {
+        "address": "0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266",
+        "private_key": "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"
+      },
+      {
+        "address": "0x70997970c51812dc3a010c7d01b50e0d17dc79c8",
+        "private_key": "0x59c6995e998f97a5a0044966f0945389dc9e86dae88c7a8412f4603b6b78690d"
```
Comment (Jenniferab5) [inline crates/cast/tests/cli/main.rs:328]: []()

File: crates/cast/bin/cmd/wallet/mod.rs [hunk 2]
```diff
@@ -308,13 +317,33 @@ impl WalletSubcommands {
...
+
+                let mut accounts = json!([]);
                 for (i, wallet) in wallets.iter().enumerate() {
-                    println!("- Account {i}:");
-                    println!("Address:     {}", wallet.address());
-                    println!("Private key: 0x{}\n", hex::encode(wallet.credential().to_bytes()));
+                    let private_key = hex::encode(wallet.credential().to_bytes());
+                    if json {
+                        accounts.as_array_mut().unwrap().push(json!({
+                            "address": wallet.address(),
```
Comment (grandizzy) [inline crates/cast/bin/cmd/wallet/mod.rs:331]: maybe do `"address": format!("{}", wallet.address()),` to preserve check-summed format

File: crates/cast/bin/cmd/wallet/mod.rs [hunk 3]
```diff
@@ -308,13 +317,33 @@ impl WalletSubcommands {
...
+                            "private_key": format!("0x{}", hex::encode(wallet.credential().to_bytes())),
```
Comment (grandizzy) [inline crates/cast/bin/cmd/wallet/mod.rs:332]:
```suggestion
                            "private_key": format!("0x{private_key}"),
```

Comment (zerosnacks) [inline crates/cast/bin/cmd/wallet/mod.rs:332]:
```suggestion
                            "private_key": format!("0x{}", private_key),
```

File: crates/cast/tests/cli/main.rs [hunk 4]
```diff
@@ -299,6 +299,42 @@ Address:     0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266
 Private key: 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80
 
 
+"#]]);
+});
+
+// tests that `cast wallet new-mnemonic --json` outputs the expected mnemonic
+casttest!(wallet_mnemonic_json, |_prj, cmd| {
```
Comment (zerosnacks) [inline crates/cast/tests/cli/main.rs:306]:
```suggestion
casttest!(wallet_mnemonic_from_entropy_json, |_prj, cmd| {
```

File: crates/cast/tests/cli/main.rs [hunk 5]
```diff
@@ -299,6 +299,42 @@ Address:     0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266
 Private key: 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80
 
 
+"#]]);
```
Comment (zerosnacks) [inline crates/cast/tests/cli/main.rs:302]: Would you mind adjusting this example to include `--accounts 3`?

File: crates/cast/bin/cmd/wallet/mod.rs [hunk 6]
```diff
@@ -290,16 +294,21 @@ impl WalletSubcommands {
                     }
                 }
             }
-            Self::NewMnemonic { words, accounts, entropy } => {
+            Self::NewMnemonic { words, accounts, entropy, json } => {
+                let mut json_values = if json { Some(vec![]) } else { None };
```
Comment (grandizzy) [inline crates/cast/bin/cmd/wallet/mod.rs:298]: don't think we need this here, we could just have at L#341
```rust
                if json {
                    let obj = json!({
                        "mnemonic": phrase,
                        "accounts": accounts,
                    });
                    println!("{}", serde_json::to_string_pretty(&obj)?);
                }
```

File: crates/cast/bin/cmd/wallet/mod.rs [hunk 7]
```diff
@@ -69,6 +69,10 @@ pub enum WalletSubcommands {
         /// Entropy to use for the mnemonic
         #[arg(long, short, conflicts_with = "words")]
         entropy: Option<String>,
+
+        /// Output generated mnemonic phrase as JSON.
```
Comment (zerosnacks) [inline crates/cast/bin/cmd/wallet/mod.rs:73]:
```suggestion
        /// Output generated mnemonic phrase and accounts as JSON.
```

**Review by grandizzy — CHANGES_REQUESTED** [review]
lgtm, couple of comments added

**Review by zerosnacks — CHANGES_REQUESTED** [review]
Supportive of the feature, some minor notes

File: crates/cast/bin/cmd/wallet/mod.rs (hunk 3 above)
Comment (leovct) [reply crates/cast/bin/cmd/wallet/mod.rs]: good catch!

File: crates/cast/tests/cli/main.rs (hunk 5 above)
Comment (leovct) [reply crates/cast/tests/cli/main.rs:317]: Addressed by 97c0226008c69872c5112f543b9e032cdfd1fe4e

File: crates/cast/bin/cmd/wallet/mod.rs (hunk 3 above)
Comment (leovct) [reply crates/cast/bin/cmd/wallet/mod.rs]: Already addressed by [Update crates/cast/bin/cmd/wallet/mod.rs](https://github.com/foundry-rs/foundry/pull/9139/commits/a803c11f119bf0d5d80cc71a8c113be5e14d52cd)

File: crates/cast/bin/cmd/wallet/mod.rs (hunk 2 above)
Comment (leovct) [reply crates/cast/bin/cmd/wallet/mod.rs]: Addressed in 6179e125b17c7ab748a0c53806c71ac1b2e99222

File: crates/cast/bin/cmd/wallet/mod.rs (hunk 6 above)
Comment (leovct) [reply crates/cast/bin/cmd/wallet/mod.rs]: Addressed in 298346d1c9150fed581d1b73e04a658349b149e3

**Review by grandizzy — APPROVED** [review]
lgtm, thanks!

**Review by zerosnacks — APPROVED** [review]
Thanks! LGTM

**Pull request merged**